### DISCUSSION TITLE
Does OpenSSL cmd-line tool support operations (asn1parse, key-gen, enc/dec, sign/verify) with quantum safe algorithms?

### DISCUSSION THREAD

**Discussion by Hussain1811** [openssl/openssl#24437] [Q&A] [answered]
Do we have support for the following quantum safe algorithms?
Dilithium (round-2, round-3 all variants)
Kyber (all variants)

**Comment by paulidale** [comment]
With the [OQS provider](https://github.com/open-quantum-safe) loaded it can.  By default it doesn't.

**Reply by Hussain1811** [reply]
So, i looked at the release notes and seems like from the 3.2 version onwards, we can load this provider. Is my understanding correct?
Also, how do i load this provider?

**Reply by paulidale** [reply]
Try reading the relevant [OQS documentation](https://github.com/open-quantum-safe/oqs-provider/blob/main/USAGE.md)?

**Reply by Hussain1811** [reply]
i've made the configuration change in `/etc/ssl/openssl.cnf`. But, `openssl list -providers` doesn't list OQS provider
```
Providers:
  default
    name: OpenSSL Default Provider
    version: 3.0.2
    status: active
```

NOTE: I haven't installed OpenSSL from sources (installed it via the package manager)

**Reply by Hussain1811** [reply]
i don't see the oqsprovider library in `/usr/lib/x86_64-linux-gnu/ossl-modules/`

**Reply by paulidale** [reply, marked as answer]
You will have to build the OQS provider from source...

**Reply by beldmit** [reply]
Or use Fedora Linux, where you can just install it

**Reply by Hussain1811** [reply]
@paulidale i was able to successfully set it up. Thank you.
